src/task/serverTaskLinkProvider.ts

```diff
@@ -146,7 +146,7 @@ export class serverTaskLinkProvider
 	): Promise<void> {
 		const hostUri = connection.constructLocalUri(connection.httpPort);
 		const extHostUri = await connection.resolveExternalHTTPUri();
-		const extHostStr = `${extHostUri.scheme}://${extHostUri.authority}`;
+		const extHostStr = encodeURI(`${extHostUri.scheme}://${extHostUri.authority}`);

```

Comment:
This just escapes characters for a URI, you need something like this: https://github.com/microsoft/vscode/blob/92ecdac13e8b955ce2069e95f47ac799e006c5d1/src/vs/base/common/strings.ts#L66